lib/Migration/Version15000Date20220809000040.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2022 Vitor Mattos <vitor@php.rio>
+ *
+ * @author Vitor Mattos <vitor@php.rio>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Talk\Migration;
+
+use Closure;
+use Doctrine\DBAL\Types\Types;
+use OCP\DB\ISchemaWrapper;
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use OCP\IDBConnection;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+
+class Version15000Date20220809000040 extends SimpleMigrationStep {
+	protected IDBConnection $connection;
+
+	public function __construct(IDBConnection $connection) {
+		$this->connection = $connection;
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 * @return null|ISchemaWrapper
+	 */
+	public function changeSchema(IOutput $output, Closure $schemaClosure, array $options): ?ISchemaWrapper {
+		/** @var ISchemaWrapper $schema */
+		$schema = $schemaClosure();
+
+		$table = $schema->getTable('talk_rooms');
+		if (!$table->hasColumn('pre_history')) {
+			// As per call the default for NEW conversations should be "Hidden".
+			$table->addColumn('pre_history', Types::BOOLEAN, [
+				'notnull' => false,
+				'default' => false
+			]);
+		}
+
+		$table = $schema->getTable('talk_attendees');
+		if (!$table->hasColumn('history_since')) {
+			$table->addColumn('history_since', Types::DATETIME_MUTABLE, [
```

Comment:
Would name this something like "joined_since" or something alike.

I wondered if we want to store the time or the message id. Don't know what would be more handy later on.

Reply:
> I wondered if we want to store the time or the message id. Don't know what would be more handy later on.

About using ID:
* If the group have "expire message" enabled, the join message will be deleted, and we will have an ID pointing to an invalid message.

About using time:
* Is clearer to identify the date limit in attendee table/instance and if we have a date time, is because this user can't will get messages before this date, if we don't have, is because we won't stop to delivery old messages.

I implemented the logic to cut the message in a loop that we do to organize the message list and when we found a message greater than join date, I break the loop and follow to delivery the previous parsed messages.

I implemented in the loop because to cut in query will be nessary to change https://github.com/nextcloud/server/blob/master/lib/private/Comments/Manager.php#L471 to define the filter as greater than the ID (or date) of join message when we have this information. This will put more complexity in this method, and I think that this maybe is unnecessary.

Reply:
Okay, so we use the time point. Should be okay.


Any reason to use DateTime instead of bigint like we use for other time points across talk?

Reply:
More human-readable only looking in the database without necessity to do conversions and because I identified other points that use DateTime instead of bigint.

Reply:
Looking the result of tests I think that we will need to use the ID.

https://drone.nextcloud.com/nextcloud/spreed/9390/2/4
https://drone.nextcloud.com/nextcloud/spreed/9388/2/4
https://drone.nextcloud.com/nextcloud/spreed/9387/2/4

This isn't different failures in different tests, is the same failure:
- I used >= to filter the history using time. But, if we have two actions in the same second of join date, the new user will see the two messages.
If we only use >, the solution to this problem, the join message will never be displayed.
I don't like to use ID in this case, as I explained before, but I think is the solution at this moment.

Reply:
Will be good to think more. I think that the solution will be to store both data.